lib/authenticator.rb

```diff
@@ -0,0 +1,29 @@
+require 'jwt'
+require 'openssl'
+
+class Authenticator
+  def self.key
+    return @key if @key
+    @key = OpenSSL::PKey.read File.read key_path
+  end
+
+  def self.from_token(token)
+    decode(token)&.dig 0, 'data'
```

Comment:
Because it didn't have those features the last time I built something like this and it is entirely undocumented. I did end up in the source for some other reason and noticed that PR that was merged in March that added a few of these convenience methods, but this was almost done at that point.

Reply:
The library methods were added as work in caesar in march, https://github.com/zooniverse/caesar/issues/598

And this tove code was written in march? 

So.... the plan is to avoid using the tested / approved library code, instead you'll roll your own to handle this?

Reply:
I'm going to open an issue regarding a refactor.